### PR TITLE
Vic 637 - making logstash appender async and buffered

### PR DESCRIPTION
**Pull request by tkuzynow**

### CODE REVIEW

File: src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java [hunk 1]
```diff
@@ -91,4 +129,12 @@ private void logToStandardError(String msg) {
   public void setEncoder(Encoder<T> encoder) {
     this.encoder = encoder;
   }
+
+  public void setLogBufferSize(int logBufferSize) {
```
Comment (mebo4b) [inline src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java:133]: Are the setter methods only for unit tests? Could you please remove them from production code and use REflectionTestUtils.setField() instead?

File: src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java [hunk 2]
```diff
@@ -1,42 +1,80 @@
...
 
-    String json = serializeToJson((T) event);
+    serializeEventAndAddToBuffer((T) event);
+    if (isBufferSizeExceeded()) {
+      sendRequestsAsynchronously();
+    }
+  }
+
+  private boolean isBufferSizeExceeded() {
+    return bufferedJsonLogRequests.size() % logBufferSize == 0;
```
[the remark this answers is not in the document]
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java:47]: yes, the intention was to send it when log exceeds the buffer size, which is configurable in logback-spring.xml (default value set to 20 though)

File: src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java (hunk 1 above)
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java:133]: they are not just for unit tests, as we want to be able to specify them in the logstash-spring.xml

File: src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java [hunk 3]
```diff
@@ -1,42 +1,80 @@
...
+  }
+
+  private void sendRequestsAsynchronously() {
```
[the remark this answers is not in the document]
Comment (tkuzynow) [reply src/main/java/de/caritas/cob/userservice/api/helper/LogstashAppender.java:63]: done

**Pull request merged**